Unexpected output while running "ListBuckets"

I have installed  minio-dotnet client library & provided it AWS and Play server credentials. I got following error.

```
$ vim Minio.Examples/ListBuckets.cs $ mcs /r:Minio/bin/Release/Minio.dll Minio.Examples/ListBuckets.cs
$ mono Minio.Examples/ListBuckets.exe

Unhandled Exception:
System.InvalidOperationException: There was an error reflecting type 'Minio.Xml.ListAllMyBucketsResult'. ---> System.InvalidOperationException: Cannot serialize member 'Minio.Xml.ListAllMyBucketsResult.Buckets' of type 'System.Collections.Generic.IReadOnlyCollection`1[[Minio.Xml.Bucket, Minio, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]]', see inner exception for more details. ---> System.NotSupportedException: Cannot serialize member Minio.Xml.ListAllMyBucketsResult.Buckets of type System.Collections.Generic.IReadOnlyCollection`1[[Minio.Xml.Bucket, Minio, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]] because it is an interface.
  --- End of inner exception stack trace ---
  at System.Xml.Serialization.StructModel.CheckSupportedMember (System.Xml.Serialization.TypeDesc typeDesc, System.Reflection.MemberInfo member, System.Type type) <0x41b25020 + 0x00273> in <filename unknown>:0
  at System.Xml.Serialization.StructModel.GetPropertyModel (System.Reflection.PropertyInfo propertyInfo) <0x41b24e20 + 0x000ef> in <filename unknown>:0
  at System.Xml.Serialization.StructModel.GetFieldModel (System.Reflection.MemberInfo memberInfo) <0x41b24ca0 + 0x000df> in <filename unknown>:0
  at System.Xml.Serialization.XmlReflectionImporter.InitializeStructMembers (System.Xml.Serialization.StructMapping mapping, System.Xml.Serialization.StructModel model, Boolean openModel, System.String typeName, System.Xml.Serialization.RecursionLimiter limiter) <0x41b23000 + 0x00943> in <filename unknown>:0
  at System.Xml.Serialization.XmlReflectionImporter.ImportStructLikeMapping (System.Xml.Serialization.StructModel model, System.String ns, Boolean openModel, System.Xml.Serialization.XmlAttributes a, System.Xml.Serialization.RecursionLimiter limiter) <0x41b20620 + 0x003db> in <filename unknown>:0
  at System.Xml.Serialization.XmlReflectionImporter.ImportTypeMapping (System.Xml.Serialization.TypeModel model, System.String ns, ImportContext context, System.String dataType, System.Xml.Serialization.XmlAttributes a, Boolean repeats, Boolean openModel, System.Xml.Serialization.RecursionLimiter limiter) <0x41b1f6f0 + 0x0098f> in <filename unknown>:0
  --- End of inner exception stack trace ---
  at System.Xml.Serialization.XmlReflectionImporter.ImportTypeMapping (System.Xml.Serialization.TypeModel model, System.String ns, ImportContext context, System.String dataType, System.Xml.Serialization.XmlAttributes a, Boolean repeats, Boolean openModel, System.Xml.Serialization.RecursionLimiter limiter) <0x41b1f6f0 + 0x00ccf> in <filename unknown>:0
  at System.Xml.Serialization.XmlReflectionImporter.ImportTypeMapping (System.Xml.Serialization.TypeModel model, System.String ns, ImportContext context, System.String dataType, System.Xml.Serialization.XmlAttributes a, System.Xml.Serialization.RecursionLimiter limiter) <0x41b1f680 + 0x00057> in <filename unknown>:0
  at System.Xml.Serialization.XmlReflectionImporter.ImportElement (System.Xml.Serialization.TypeModel model, System.Xml.Serialization.XmlRootAttribute root, System.String defaultNamespace, System.Xml.Serialization.RecursionLimiter limiter) <0x41b1d8f0 + 0x000d7> in <filename unknown>:0
  at System.Xml.Serialization.XmlReflectionImporter.ImportTypeMapping (System.Type type, System.Xml.Serialization.XmlRootAttribute root, System.String defaultNamespace) <0x41b1adc0 + 0x0009b> in <filename unknown>:0
  at System.Xml.Serialization.XmlSerializer..ctor (System.Type type, System.String defaultNamespace) <0x41b01eb0 + 0x0036f> in <filename unknown>:0
  at System.Xml.Serialization.XmlSerializer..ctor (System.Type type) <0x41b01d70 + 0x00013> in <filename unknown>:0
  at Minio.MinioClient.ListBuckets () <0x41a6e4a0 + 0x00183> in <filename unknown>:0
  at Minio.Examples.ListBuckets.Main () <0x41a4fd50 + 0x00067> in <filename unknown>:0
[ERROR] FATAL UNHANDLED EXCEPTION: System.InvalidOperationException: There was an error reflecting type 'Minio.Xml.ListAllMyBucketsResult'. ---> System.InvalidOperationException: Cannot serialize member 'Minio.Xml.ListAllMyBucketsResult.Buckets' of type 'System.Collections.Generic.IReadOnlyCollection`1[[Minio.Xml.Bucket, Minio, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]]', see inner exception for more details. ---> System.NotSupportedException: Cannot serialize member Minio.Xml.ListAllMyBucketsResult.Buckets of type System.Collections.Generic.IReadOnlyCollection`1[[Minio.Xml.Bucket, Minio, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]] because it is an interface.
  --- End of inner exception stack trace ---
  at System.Xml.Serialization.StructModel.CheckSupportedMember (System.Xml.Serialization.TypeDesc typeDesc, System.Reflection.MemberInfo member, System.Type type) <0x41b25020 + 0x00273> in <filename unknown>:0
  at System.Xml.Serialization.StructModel.GetPropertyModel (System.Reflection.PropertyInfo propertyInfo) <0x41b24e20 + 0x000ef> in <filename unknown>:0
  at System.Xml.Serialization.StructModel.GetFieldModel (System.Reflection.MemberInfo memberInfo) <0x41b24ca0 + 0x000df> in <filename unknown>:0
  at System.Xml.Serialization.XmlReflectionImporter.InitializeStructMembers (System.Xml.Serialization.StructMapping mapping, System.Xml.Serialization.StructModel model, Boolean openModel, System.String typeName, System.Xml.Serialization.RecursionLimiter limiter) <0x41b23000 + 0x00943> in <filename unknown>:0
  at System.Xml.Serialization.XmlReflectionImporter.ImportStructLikeMapping (System.Xml.Serialization.StructModel model, System.String ns, Boolean openModel, System.Xml.Serialization.XmlAttributes a, System.Xml.Serialization.RecursionLimiter limiter) <0x41b20620 + 0x003db> in <filename unknown>:0
  at System.Xml.Serialization.XmlReflectionImporter.ImportTypeMapping (System.Xml.Serialization.TypeModel model, System.String ns, ImportContext context, System.String dataType, System.Xml.Serialization.XmlAttributes a, Boolean repeats, Boolean openModel, System.Xml.Serialization.RecursionLimiter limiter) <0x41b1f6f0 + 0x0098f> in <filename unknown>:0

▽
 * limitations under the License.
  --- End of inner exception stack trace ---
  at System.Xml.Serialization.XmlReflectionImporter.ImportTypeMapping (System.Xml.Serialization.TypeModel model, System.String ns, ImportContext context, System.String dataType, System.Xml.Serialization.XmlAttributes a, Boolean repeats, Boolean openModel, System.Xml.Serialization.RecursionLimiter limiter) <0x41b1f6f0 + 0x00ccf> in <filename unknown>:0
  at System.Xml.Serialization.XmlReflectionImporter.ImportTypeMapping (System.Xml.Serialization.TypeModel model, System.String ns, ImportContext context, System.String dataType, System.Xml.Serialization.XmlAttributes a, System.Xml.Serialization.RecursionLimiter limiter) <0x41b1f680 + 0x00057> in <filename unknown>:0
  at System.Xml.Serialization.XmlReflectionImporter.ImportElement (System.Xml.Serialization.TypeModel model, System.Xml.Serialization.XmlRootAttribute root, System.String defaultNamespace, System.Xml.Serialization.RecursionLimiter limiter) <0x41b1d8f0 + 0x000d7> in <filename unknown>:0
  at System.Xml.Serialization.XmlReflectionImporter.ImportTypeMapping (System.Type type, System.Xml.Serialization.XmlRootAttribute root, System.String defaultNamespace) <0x41b1adc0 + 0x0009b> in <filename unknown>:0
  at System.Xml.Serialization.XmlSerializer..ctor (System.Type type, System.String defaultNamespace) <0x41b01eb0 + 0x0036f> in <filename unknown>:0
  at System.Xml.Serialization.XmlSerializer..ctor (System.Type type) <0x41b01d70 + 0x00013> in <filename unknown>:0
  at Minio.MinioClient.ListBuckets () <0x41a6e4a0 + 0x00183> in <filename unknown>:0
  at Minio.Examples.ListBuckets.Main () <0x41a4fd50 + 0x00067> in <filename unknown>:0
$
$ vim Minio.Examples/ListBuckets.cs
$ mcs /r:Minio/bin/Release/Minio.dll Minio.Examples/ListBuckets.cs
$ export MONO_PATH=Minio/bin/Release
$ mono Minio.Examples/ListBuckets.exe

Unhandled Exception:
System.InvalidOperationException: There was an error reflecting type 'Minio.Xml.ListAllMyBucketsResult'. ---> System.InvalidOperationException: Cannot serialize member 'Minio.Xml.ListAllMyBucketsResult.Buckets' of type 'System.Collections.Generic.IReadOnlyCollection`1[[Minio.Xml.Bucket, Minio, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]]', see inner exception for more details. ---> System.NotSupportedException: Cannot serialize member Minio.Xml.ListAllMyBucketsResult.Buckets of type System.Collections.Generic.IReadOnlyCollection`1[[Minio.Xml.Bucket, Minio, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]] because it is an interface.
  --- End of inner exception stack trace ---
  at System.Xml.Serialization.StructModel.CheckSupportedMember (System.Xml.Serialization.TypeDesc typeDesc, System.Reflection.MemberInfo member, System.Type type) <0x409af020 + 0x00273> in <filename unknown>:0
  at System.Xml.Serialization.StructModel.GetPropertyModel (System.Reflection.PropertyInfo propertyInfo) <0x409aee20 + 0x000ef> in <filename unknown>:0
  at System.Xml.Serialization.StructModel.GetFieldModel (System.Reflection.MemberInfo memberInfo) <0x409aeca0 + 0x000df> in <filename unknown>:0
  at System.Xml.Serialization.XmlReflectionImporter.InitializeStructMembers (System.Xml.Serialization.StructMapping mapping, System.Xml.Serialization.StructModel model, Boolean openModel, System.String typeName, System.Xml.Serialization.RecursionLimiter limiter) <0x409ad000 + 0x00943> in <filename unknown>:0
  at System.Xml.Serialization.XmlReflectionImporter.ImportStructLikeMapping (System.Xml.Serialization.StructModel model, System.String ns, Boolean openModel, System.Xml.Serialization.XmlAttributes a, System.Xml.Serialization.RecursionLimiter limiter) <0x409abc90 + 0x003db> in <filename unknown>:0
  at System.Xml.Serialization.XmlReflectionImporter.ImportTypeMapping (System.Xml.Serialization.TypeModel model, System.String ns, ImportContext context, System.String dataType, System.Xml.Serialization.XmlAttributes a, Boolean repeats, Boolean openModel, System.Xml.Serialization.RecursionLimiter limiter) <0x409aad60 + 0x0098f> in <filename unknown>:0
  --- End of inner exception stack trace ---
  at System.Xml.Serialization.XmlReflectionImporter.ImportTypeMapping (System.Xml.Serialization.TypeModel model, System.String ns, ImportContext context, System.String dataType, System.Xml.Serialization.XmlAttributes a, Boolean repeats, Boolean openModel, System.Xml.Serialization.RecursionLimiter limiter) <0x409aad60 + 0x00ccf> in <filename unknown>:0
  at System.Xml.Serialization.XmlReflectionImporter.ImportTypeMapping (System.Xml.Serialization.TypeModel model, System.String ns, ImportContext context, System.String dataType, System.Xml.Serialization.XmlAttributes a, System.Xml.Serialization.RecursionLimiter limiter) <0x409aacf0 + 0x00057> in <filename unknown>:0
  at System.Xml.Serialization.XmlReflectionImporter.ImportElement (System.Xml.Serialization.TypeModel model, System.Xml.Serialization.XmlRootAttribute root, System.String defaultNamespace, System.Xml.Serialization.RecursionLimiter limiter) <0x409a8f60 + 0x000d7> in <filename unknown>:0
  at System.Xml.Serialization.XmlReflectionImporter.ImportTypeMapping (System.Type type, System.Xml.Serialization.XmlRootAttribute root, System.String defaultNamespace) <0x409a6430 + 0x0009b> in <filename unknown>:0
  at System.Xml.Serialization.XmlSerializer..ctor (System.Type type, System.String defaultNamespace) <0x4099ddc0 + 0x0036f> in <filename unknown>:0
  at System.Xml.Serialization.XmlSerializer..ctor (System.Type type) <0x4099dc80 + 0x00013> in <filename unknown>:0
  at Minio.MinioClient.ListBuckets () <0x4090cd70 + 0x00183> in <filename unknown>:0
  at Minio.Examples.ListBuckets.Main () <0x408d9d50 + 0x00067> in <filename unknown>:0
[ERROR] FATAL UNHANDLED EXCEPTION: System.InvalidOperationException: There was an error reflecting type 'Minio.Xml.ListAllMyBucketsResult'. ---> System.InvalidOperationException: Cannot serialize member 'Minio.Xml.ListAllMyBucketsResult.Buckets' of type 'System.Collections.Generic.IReadOnlyCollection`1[[Minio.Xml.Bucket, Minio, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]]', see inner exception for more details. ---> System.NotSupportedException: Cannot serialize member Minio.Xml.ListAllMyBucketsResult.Buckets of type System.Collections.Generic.IReadOnlyCollection`1[[Minio.Xml.Bucket, Minio, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]] because it is an interface.
  --- End of inner exception stack trace ---
  at System.Xml.Serialization.StructModel.CheckSupportedMember (System.Xml.Serialization.TypeDesc typeDesc, System.Reflection.MemberInfo member, System.Type type) <0x409af020 + 0x00273> in <filename unknown>:0
  at System.Xml.Serialization.StructModel.GetPropertyModel (System.Reflection.PropertyInfo propertyInfo) <0x409aee20 + 0x000ef> in <filename unknown>:0
  at System.Xml.Serialization.StructModel.GetFieldModel (System.Reflection.MemberInfo memberInfo) <0x409aeca0 + 0x000df> in <filename unknown>:0
  at System.Xml.Serialization.XmlReflectionImporter.InitializeStructMembers (System.Xml.Serialization.StructMapping mapping, System.Xml.Serialization.StructModel model, Boolean openModel, System.String typeName, System.Xml.Serialization.RecursionLimiter limiter) <0x409ad000 + 0x00943> in <filename unknown>:0
  at System.Xml.Serialization.XmlReflectionImporter.ImportStructLikeMapping (System.Xml.Serialization.StructModel model, System.String ns, Boolean openModel, System.Xml.Serialization.XmlAttributes a, System.Xml.Serialization.RecursionLimiter limiter) <0x409abc90 + 0x003db> in <filename unknown>:0
  at System.Xml.Serialization.XmlReflectionImporter.ImportTypeMapping (System.Xml.Serialization.TypeModel model, System.String ns, ImportContext context, System.String dataType, System.Xml.Serialization.XmlAttributes a, Boolean repeats, Boolean openModel, System.Xml.Serialization.RecursionLimiter limiter) <0x409aad60 + 0x0098f> in <filename unknown>:0

▽

  --- End of inner exception stack trace ---
  at System.Xml.Serialization.XmlReflectionImporter.ImportTypeMapping (System.Xml.Serialization.TypeModel model, System.String ns, ImportContext context, System.String dataType, System.Xml.Serialization.XmlAttributes a, Boolean repeats, Boolean openModel, System.Xml.Serialization.RecursionLimiter limiter) <0x409aad60 + 0x00ccf> in <filename unknown>:0
  at System.Xml.Serialization.XmlReflectionImporter.ImportTypeMapping (System.Xml.Serialization.TypeModel model, System.String ns, ImportContext context, System.String dataType, System.Xml.Serialization.XmlAttributes a, System.Xml.Serialization.RecursionLimiter limiter) <0x409aacf0 + 0x00057> in <filename unknown>:0
  at System.Xml.Serialization.XmlReflectionImporter.ImportElement (System.Xml.Serialization.TypeModel model, System.Xml.Serialization.XmlRootAttribute root, System.String defaultNamespace, System.Xml.Serialization.RecursionLimiter limiter) <0x409a8f60 + 0x000d7> in <filename unknown>:0
  at System.Xml.Serialization.XmlReflectionImporter.ImportTypeMapping (System.Type type, System.Xml.Serialization.XmlRootAttribute root, System.String defaultNamespace) <0x409a6430 + 0x0009b> in <filename unknown>:0

▽
  at System.Xml.Serialization.XmlSerializer..ctor (System.Type type, System.String defaultName
▽
using System.IO;
space) <0x4099ddc0 + 0x0036f> in <filename unknown>:0
  at System.Xml.Serialization.XmlSerializer..ctor (System.Type type) <0x4099dc80 + 0x00013> in <filename unknown>:0
  at Minio.MinioClient.ListBuckets () <0x4090cd70 + 0x00183> in <filename unknown>:0
  at Minio.Examples.ListBuckets.Main () <0x408d9d50 + 0x00067> in <filename unknown>:0
```

--- a/Minio/Minio.csproj
+++ b/Minio/Minio.csproj
@@ -1,4 +1,4 @@
-<?xml version="1.0" encoding="utf-8"?>
+﻿<?xml version="1.0" encoding="utf-8"?>
 <Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" Condition="Exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props')" />
   <PropertyGroup>
@@ -117,6 +117,9 @@
       <Generator>ResXFileCodeGenerator</Generator>
       <LastGenOutput>Resources.Designer.cs</LastGenOutput>
     </EmbeddedResource>
+  </ItemGroup>
+  <ItemGroup>
+    <Content Include="Scripts\ListBuckets.fsx" />
   </ItemGroup>
   <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
   <!-- To modify your build process, add your task inside one of the targets below and uncomment it.

--- a/Minio/MinioClient.cs
+++ b/Minio/MinioClient.cs
@@ -433,7 +433,7 @@ namespace Minio
         /// Lists all buckets owned by the user
         /// </summary>
         /// <returns>A list of all buckets owned by the user.</returns>
-        public IReadOnlyCollection<Bucket> ListBuckets()
+        public List<Bucket> ListBuckets()
         {
             var request = new RestRequest("/", Method.GET);
             var response = client.Execute(request);

--- a/Minio/Scripts/ListBuckets.fsx
+++ b/Minio/Scripts/ListBuckets.fsx
@@ -5,5 +5,5 @@ open Minio
 open System
 
 /// N B: Listing buckets on minio server at play.minio.io test access-key and secret-key
-let mc = new MinioClient("https://play.minio.io:9000", "Q3AM3UQ867SPQQA43P2F",  "zuf+tfteSlswRu7BJ86wekitnifILbZam1KYY3TG")
-mc.ListBuckets() |> Seq.iter (fun b -> printfn "%s" b.Name)
+let minioClient = new MinioClient("https://play.minio.io:9000", "Q3AM3UQ867SPQQA43P2F",  "zuf+tfteSlswRu7BJ86wekitnifILbZam1KYY3TG")
+minioClient.ListBuckets() |> Seq.iter (fun b -> printfn "%s" b.Name)

--- a/Minio/Scripts/ListBuckets.fsx
+++ b/Minio/Scripts/ListBuckets.fsx
@@ -1,0 +1,9 @@
+#I __SOURCE_DIRECTORY__
+#r "../bin/Release/Minio.dll"
+
+open Minio
+open System
+
+/// N B: Listing buckets on minio server at play.minio.io test access-key and secret-key
+let mc = new MinioClient("https://play.minio.io:9000", "Q3AM3UQ867SPQQA43P2F",  "zuf+tfteSlswRu7BJ86wekitnifILbZam1KYY3TG")
+mc.ListBuckets() |> Seq.iter (fun b -> printfn "%s" b.Name)

--- a/Minio/Xml/ListAllMyBucketsResult.cs
+++ b/Minio/Xml/ListAllMyBucketsResult.cs
@@ -30,6 +30,6 @@ namespace Minio.Xml
 
         [XmlArray("Buckets")]
         [XmlArrayItem(typeof(Bucket))]
-        public IReadOnlyCollection<Bucket> Buckets { get; set; }
+        public List<Bucket> Buckets { get; set; }
     }
 }

